radix-engine/src/wasm/wasm_instrumenter.rs

```diff
@@ -28,17 +30,9 @@ pub struct InstrumentedCode {
 }
 
 impl WasmInstrumenter {
-    pub fn new(options: InstrumenterOptions) -> Self {
-        let cache = Cache::builder()
-            .weigher(|_key: &(Hash, Hash), value: &Arc<Vec<u8>>| -> u32 {
-                value
-                    .len()
-                    .checked_add(Hash::LENGTH * 2)
-                    .and_then(|total| total.try_into().ok())
-                    .unwrap_or(u32::MAX)
-            })
-            .max_capacity(options.max_cache_size_bytes)
-            .build();
+    pub fn new(_options: InstrumenterOptions) -> Self {
+        // TODO: limit size
+        let cache = RefCell::new(HashMap::new());
```

Comment:
This feels like a definite reversion of behaviour. We want a safe and performant cache implementation here, for use in the node.

Can we instead just fall back to a RefCell / HashMap in the no-std case?

Reply:
Better yet, what about creating our own `ThreadsafeCache` type, and swap out the implementation for either wrapping Moka or a Mutex<HashMap<K, Arc<X>>> depending on if we use `alloc` or not?

Reply:
`Mutex` is not part of the `no_std`.